openapi.yaml

```
@@ -75,6 +77,127 @@ paths:
                     type: string
               required:
                 - file
+  /status_report/{analysis_id}:
+    get:
+      tags:
+        - Status report
+      summary: Get analysis status report
+      description: >
+        Retrieves the report of a firmware analysis.
+
+        - If the analysis is still **running**, returns its progress
+
+        - If the analysis has **failed**, returns failure message, queues zip file generation. Returns the download link on subsequent request
+
+        - If analysis is **finished**, queues zip file generation. Returns the download link on subsequent request
+
+      operationId: getStatusReport
+      parameters:
+        - name: analysis_id
+          in: path
+          required: true
+          description: UUID of the analysis
+          schema:
+            type: string
+            format: uuid
+        - name: api_key
+          in: query
+          required: true
+          description: API key for authentication
+          schema:
+            type: string
```

```suggestion
security:
        - ApiKeyAuth: []
```
This is not part of the parent issue. Rather state that API auth is required.

---

Right now, the `api_key` is included in the URL.

Since the other pull request is merged, I'll change it and put the `api_key` in the header and update the documentation.